low-level/framework/utils/salt_util.py

```diff
@@ -13,88 +13,114 @@
-        node_key = 'srvnode-1'
+        '''init method'''
+        if CommonConfig.__instance is None:
+            self.utility = Utility()
+            self.pillar_info = None
+            _result, _err, _ret_code = \
+                    self.utility.execute_cmd(['sudo', 'provisioner', 'pillar_get'])
+            _err = _err.decode("utf-8").rstrip()
+            if _ret_code == 0 and _err is '':
+                self.pillar_info = eval(_result.decode("utf-8").rstrip())
+            self.node_id = None or self.get_node_id()
+            self._is_single_node = None or self._is_single_node()
+            self.consul_host = None or self.get_consul_vip()
+            self.consul_port = None or self.get_consul_port()
+            CommonConfig.__instance = self
+
+    @staticmethod
+    def get_singleton_instance():
+        '''Returns an instance of this class'''
+        if CommonConfig.__instance is None:
+            CommonConfig()
+        return CommonConfig.__instance
+
+    @staticmethod
+    def get_node_id():
+        '''Returns salt minion_id using salt config file'''
+
+        _node_id = 'srvnode-1'
         try:
-            subout = subprocess.Popen(self.GRAINS_GET_NODE_CMD, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            result = subout.stdout.readlines()
-            if result == [] or result == "":
-                logger.warning(f"Command '{self.GRAINS_GET_NODE_CMD}' failed to fetch grain id or hostname.")
-                subout = subprocess.Popen(self.MINION_GET_NODE_CMD, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-                result = subout.stdout.readlines()
-                if result == [] or result == "":
-                    logger.warning(f"Command '{self.MINION_GET_NODE_CMD}' failed to fetch minion id or hostname.")
-                    logger.warning("Using node_id ('srvnode-1') as we are not able to fetch it from hostname command.")
-                    node_key = 'srvnode-1'
-                else:
-                    node_key = result[0].decode().rstrip('\n').lower()
-            else:
-                node_key = result[0].decode().rstrip('\n')
-        except Exception as e:
-            logger.warning(f"Can't read node id, using 'srvnode-1' : {e}")
-        return node_key
+            with open('/etc/salt/minion_id', 'r') as salt_file:
+                _node_id = salt_file.read()
+            if _node_id is not None:
+                _node_id = 'srvnode-1'
+        except Exception as err:
+            logger.warning(f'sspl_constants, Fail to read node-id with error: {err}, \
+                             Keeping default value as srvnode-1')
+            _node_id = 'srvnode-1'
+        return _node_id
 
-    def is_single_node(self):
+    def _is_single_node(self):
         """
         Returns true if single node, false otherwise
         """
-        is_single_node = True
+        _is_single_node = False
```

Comment:
same name used for function & var. Now when we have the chance to update, use some other name

Reply:
Done.